### PR TITLE
CMP: Extend revocation request with issuer and serial

### PR DESCRIPTION
**Pull request by rajeev-0**
CMP: support specifying certificate to be revoked via issuer and serial number

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by DDvO** [comment]
Ping for reviews

**Comment by DDvO** [comment]
Can we please get this done soon, such that it surely ends up in 3.2?
Our customer paying for this asked again recently on the status.

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
Asking @openssl/committers for 2nd approval.

File: apps/cmp.c [hunk 1]
```diff
@@ -1598,12 +1602,44 @@ static int setup_request_ctx(OSSL_CMP_CTX *ctx, ENGINE *engine)
...
-            return 0;
+        if (opt_issuer == NULL && opt_serial == NULL) {
+            if (opt_oldcert == NULL && opt_csr == NULL) {
+                CMP_err("missing -oldcert or -issuer and -serial for certificate to be revoked and no -csr given");
+                return 0;
+            }
+            if (opt_oldcert != NULL && opt_csr != NULL)
+                CMP_warn("ignoring -csr since certificate to be revoked is given");
+        } else {
+#define OSSL_CMP_RR_MSG "since -issuer and -serial is given for command 'rr'"
```
Comment (beldmit) [inline apps/cmp.c:1613]: I personally don't like `#define` in the middle of the code

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by DDvO** [comment]
Merged - thanks @t8m and @beldmit for your approvals.

File: apps/cmp.c (hunk 1 above)
Comment (DDvO) [reply apps/cmp.c:1613]: > I personally don't like `#define` in the middle of the code

I agree it looks uncommon. Yet I've checked again in https://www.openssl.org/policies/technical/coding-style.html and cannot find requirements there where to place `#define` directives.
Maybe we should add something like this there and to `util/check-format.pl`.

Comment (DDvO) [reply apps/cmp.c:1613]: Added a respective candidate entry to the "Misc stuff" list in #10725.

**Pull request closed without merging**